week-1/Homework/mandatory/5-magic-8-ball.js

```diff
@@ -43,19 +43,69 @@ Outlook not so good.
 Very doubtful.
 */
 
+let array = [
+  // Very Positive:
+  "It is certain",
+  "It is decidedly so",
+  "Without a doubt",
+  "Yes - definitely",
+  "You may rely on it",
+  // Positive:
+  "As I see it, yes",
+  "Most likely",
+  "Outlook good",
+  "Yes",
+  "Signs point to yes",
+  // Negative:
+  "Reply hazy, try again",
+  "Ask again later",
+  "Better not tell you now",
+  "Cannot predict now",
+  "Concentrate and ask again.",
+  // very Negative:
+  "Don't count on it",
+  "My reply is no",
+  "My sources say no",
+  "Outlook not so good",
+  "Very doubtful."
+];
+
+function randomAnswer(array) {
+  return array[Math.floor(Math.random() * array.length)];
+}
+console.log(randomAnswer(array));
+
 // This should log "The ball has shaken!"
 // and return the answer.
-function shakeBall() {}
+function shakeBall() {
+  const answer = randomAnswer(array);
+  console.log("The ball has shaken!");
+  return answer;
+}
 
 // The answer should come from shaking the ball
-let answer;
+// let answer;
 
 // When checking the answer, we should tell someone if the answer is
 // - very positive
 // - positive
 // - negative
 // - very negative
-function checkAnswer() {}
+let answerType;
+
+function checkAnswer(answer) {
+  if (array.indexOf(randomAnswer(array)) < 5) {
+    answerType = "very positive";
+  } else if (array.indexOf(randomAnswer(array)) < 10) {
+    answerType = "positive";
+  } else if (array.indexOf(randomAnswer(array)) < 15) {
+    answerType = "negative";
+  } else {
+    answerType = "very negative";
+  }
+  // console.log(answerType);
+  return answerType;
```

Comment:
Did you intend to modify `answer` instead of `answerType` here? `answerType` is a global variable which you're modifying, instead of `answer`.

Reply:
(answer) is the argument for checkAnswer function. 
(answerType) is what checkAnswer function returns i.e. "very positive", "positive" etc... 
I declared it globally in order to use it in the function.. 
Is this bad practise? 
Should I have declared it within the function instead?

Reply:
The function returns the right kind of result (a string containing an answer type) but that answer type should be based on the `answer` parameter provided to the function.